packages/block-editor/src/components/border-radius-control/single-input-control.js

```diff
@@ -73,14 +73,7 @@ export default function SingleInputControl( {
 
 	const onChangeUnit = ( next ) => {
 		const newUnits = { ...selectedUnits };
-		if ( corner === 'all' ) {
-			newUnits.topLeft = next;
-			newUnits.topRight = next;
-			newUnits.bottomLeft = next;
-			newUnits.bottomRight = next;
-		} else {
```

Comment:
I think we might still want to retain this `if` block, but just move `newUnits[ corner ] = next;` out from the `else`. Otherwise, what's happening in this PR is that if I don't have a value set, and I change the unit, then when I expand the controls, they revert to `px` instead of inheriting from the parent / all control:

https://github.com/user-attachments/assets/da6d46c4-75b0-4b30-88d6-b1c131780a41

Reply:
Perhaps we could leave the conditional statement unchanged and just explicitly update the `flat` property?

```javascript
const onChangeUnit = ( next ) => {
	const newUnits = { ...selectedUnits };
	if ( corner === 'all' ) {
		newUnits.flat = next;
		newUnits.topLeft = next;
		newUnits.topRight = next;
		newUnits.bottomLeft = next;
		newUnits.bottomRight = next;
	} else {
		newUnits[ corner ] = next;
	}
	setSelectedUnits( newUnits );
};
```


https://github.com/user-attachments/assets/69eeb72f-1655-4628-9e50-2a587eb818d9